### PR TITLE
Add expression filter overloads to GetTables

### PR DESCRIPTION
**Pull request by christothes**
related to #19881

### CODE REVIEW

**Review by jsquire — APPROVED** [review]

File: sdk/tables/Azure.Data.Tables/src/TableClient.cs [hunk 1]
```diff
@@ -21,14 +21,36 @@ namespace Azure.Data.Tables
...
+
+        /// <summary>
+        /// The name of the table with which this client instance will interact.
+        /// </summary>
+        public string Name { get; }
+
+        /// <summary>
+        /// The name of the table account with which this client instance will interact.
+        /// </summary>
+        /// <value></value>
```
Comment (jsquire) [inline sdk/tables/Azure.Data.Tables/src/TableClient.cs:41]: nit:  If we're not filling out `<value>` , we may want to consider removing it.

File: sdk/tables/Azure.Data.Tables/api/Azure.Data.Tables.netstandard2.0.cs [hunk 2]
```diff
@@ -15,6 +15,8 @@ public partial class TableClient
         public TableClient(System.Uri endpoint, string tableName, Azure.AzureSasCredential credential, Azure.Data.Tables.TableClientOptions options = null) { }
         public TableClient(System.Uri endpoint, string tableName, Azure.Data.Tables.TableSharedKeyCredential credential) { }
         public TableClient(System.Uri endpoint, string tableName, Azure.Data.Tables.TableSharedKeyCredential credential, Azure.Data.Tables.TableClientOptions options = null) { }
+        public string AccountName { get { throw null; } }
```
Comment (pakrym) [inline sdk/tables/Azure.Data.Tables/api/Azure.Data.Tables.netstandard2.0.cs:18]: virtual

File: sdk/tables/Azure.Data.Tables/api/Azure.Data.Tables.netstandard2.0.cs [hunk 3]
```diff
@@ -97,6 +99,8 @@ public partial class TableServiceClient
         public TableServiceClient(System.Uri endpoint, Azure.AzureSasCredential credential, Azure.Data.Tables.TableClientOptions options = null) { }
         public TableServiceClient(System.Uri endpoint, Azure.Data.Tables.TableSharedKeyCredential credential) { }
         public TableServiceClient(System.Uri endpoint, Azure.Data.Tables.TableSharedKeyCredential credential, Azure.Data.Tables.TableClientOptions options = null) { }
+        public string AccountName { get { throw null; } }
```
Comment (pakrym) [inline sdk/tables/Azure.Data.Tables/api/Azure.Data.Tables.netstandard2.0.cs:102]: virtual

File: sdk/tables/Azure.Data.Tables/api/Azure.Data.Tables.netstandard2.0.cs (hunk 2 above)
Comment (christothes) [reply sdk/tables/Azure.Data.Tables/api/Azure.Data.Tables.netstandard2.0.cs:18]: Thanks - Do you think the analyzer should be flagging these?

File: sdk/tables/Azure.Data.Tables/api/Azure.Data.Tables.netstandard2.0.cs [hunk 4]
```diff
@@ -97,6 +99,8 @@ public partial class TableServiceClient
...
+        public static string CreateQueryFilter(System.Linq.Expressions.Expression<System.Func<Azure.Data.Tables.Models.TableItem, bool>> filter) { throw null; }
```
[the remark this answers is not in the document]
Comment (christothes) [reply sdk/tables/Azure.Data.Tables/api/Azure.Data.Tables.netstandard2.0.cs:103]: That one is generic, and this one is always `Expression<System.Func<TableItem, bool>>`

[the remark this answers is not in the document]
Comment (pakrym) [reply sdk/tables/Azure.Data.Tables/api/Azure.Data.Tables.netstandard2.0.cs:103]: Right, but you can use generic version with the `TableItem` as the T to get the same result.

File: sdk/tables/Azure.Data.Tables/tests/TableServiceClientLiveTests.cs [hunk 5]
```diff
@@ -228,6 +228,13 @@ public async Task GetTablesReturnsTablesWithFilter()
 
                 Assert.That(() => tableResponses, Is.Not.Empty);
                 Assert.That(() => tableResponses.Select(r => r.TableName), Contains.Item(tableName));
+
+                // Query with a filter.
+
+                tableResponses = (await service.GetTablesAsync(filter: t => t.TableName == tableName).ToEnumerableAsync().ConfigureAwait(false)).ToList();
+
+                Assert.That(() => tableResponses, Is.Not.Empty);
+                Assert.That(() => tableResponses.Select(r => r.TableName), Contains.Item(tableName));
```
Comment (pakrym) [inline sdk/tables/Azure.Data.Tables/tests/TableServiceClientLiveTests.cs:237]: Won't this condition be true if the filter is not applied at all and we get all tables back?

File: sdk/tables/Azure.Data.Tables/api/Azure.Data.Tables.netstandard2.0.cs (hunk 4 above)
[the remark this answers is not in the document]
Comment (christothes) [reply sdk/tables/Azure.Data.Tables/api/Azure.Data.Tables.netstandard2.0.cs:103]: The other benefit is that you don't have to do `TableClient.CreateQueryFilter` for a TableService query.. I don't think it hurts anything to have it here either.

**Review by pakrym — APPROVED** [review]

File: sdk/tables/Azure.Data.Tables/api/Azure.Data.Tables.netstandard2.0.cs (hunk 2 above)
Comment (pakrym) [reply sdk/tables/Azure.Data.Tables/api/Azure.Data.Tables.netstandard2.0.cs]: I don't think we have an analyzer like that.

File: sdk/tables/Azure.Data.Tables/tests/TableServiceClientLiveTests.cs (hunk 5 above)
Comment (christothes) [reply sdk/tables/Azure.Data.Tables/tests/TableServiceClientLiveTests.cs:237]: good catch - fixed.

File: sdk/tables/Azure.Data.Tables/api/Azure.Data.Tables.netstandard2.0.cs (hunk 2 above)
Comment (christothes) [reply sdk/tables/Azure.Data.Tables/api/Azure.Data.Tables.netstandard2.0.cs]: I was thinking of the one that validates that we have a sync and async version of methods. I thought it also checked that they were virtual.

**Pull request merged**